src/cogs/snipe.py

```diff
@@ -25,11 +25,18 @@ def __init__(self, bot):
         self._message_history: dict[int, list[discord.Message]] = defaultdict(lambda: deque(maxlen=conf.max_edit_msg))
 
     @commands.hybrid_command(with_app_command=True)
-    async def snipe(self, ctx: Context, number: int = 0):
-        if ctx.channel.id not in self._deleted:
-            return await ctx.send("No message to snipe.")
+    async def snipe(self, ctx: Context, number: int = 0, channel: int = 0):
```

Comment:
This does not check if the user would actually have perms to the channel they're trying to snipe.